optimum/graphcore/models/wav2vec2/modeling_wav2vec2.py

```diff
@@ -281,28 +308,189 @@ def forward(
 
     @staticmethod
     def compute_contrastive_logits(
-        target_features: torch.FloatTensor,
-        negative_features: torch.FloatTensor,
-        predicted_features: torch.FloatTensor,
-        temperature: int = 0.1,
+            target_features: torch.FloatTensor,
+            negative_features: torch.FloatTensor,
+            predicted_features: torch.FloatTensor,
+            temperature: int = 0.1,
     ):
         """
         Compute logits for contrastive loss based using cosine similarity as the distance measure between
         `[positive_feature, negative_features]` and `[predicted_features]`. Additionally, temperature can be applied.
         """
         target_features = torch.cat([target_features, negative_features], dim=0)
 
-        logits = torch.cosine_similarity(
-            predicted_features.float(), target_features.float(), dim=-1, eps=1e-4
-        ).type_as(target_features)
+        logits = torch.cosine_similarity(predicted_features, target_features, dim=-1, eps=1e-4)
```

Comment:
Do we definitely not need the casts here?

Reply:
If its BASE we can put them back in. For LARGE there's no way it will compile without half-precision or a serious change in the loss forward-backward

Reply:
In which case; I think this should be controlled by an argument so it can be switched in the config files.